api/src/main/java/jakarta/data/expression/literal/TemporalLiteralRecord.java

```diff
@@ -34,12 +35,21 @@ record TemporalLiteralRecord<T, V extends Temporal & Comparable<? extends Tempor
 
     @Override
     public String toString() {
-        return switch (value) {
-            case Instant i       -> "TIMESTAMP('" + i + "')";
-            case LocalDateTime d -> "TIMESTAMP('" + d + "')";
-            case LocalDate d     -> "DATE('" + d + "')";
-            case LocalTime t     -> "TIME('" + t + "')";
-            default              -> value.toString();
+        Temporal temporal = value instanceof Instant i
+                ? i.atOffset(ZoneOffset.UTC).toLocalDateTime()
+                : value;
+
+        return switch (temporal) {
+            case LocalDateTime d ->
+                "{ts '" + d.toLocalDate() + ' ' + d.toLocalTime() + "'}";
+            case LocalDate d ->
+                "{d '" + value + "'}";
+            case LocalTime t ->
+                "{t '" + value + "'}";
+            default ->
+                "{TemporalLiteral '"
+                        + value.getClass().getName() + " '"
+                        + value + "'}";
```

Comment:
This will probably need to be revisited in light of https://github.com/jakartaee/persistence/issues/453.  (Currently an issue in the Persistence project, but really belongs in Query.)

Reply:
> This will probably need to be revisited in light of [jakartaee/persistence#453](https://github.com/jakartaee/persistence/issues/453). (Currently an issue in the Persistence project, but really belongs in Query.)

Yes, definitely. I noticed Jakarta Persistence 3.2 was very sparse on this topic and only deferred to JDBC escape syntax for timestamp/date/time. I'm glad to see the issue open for it, and I'll get a corresponding issue opened against Jakarta Data to align with whatever the outcome is.